### PR TITLE
Mark System.Security.Cryptography.Algorithms as unsupported on Browser WASM

### PR DESCRIPTION
**Pull request by MaximLipnin**
There are some types from `System.Security.Cryptography.Algorithms` which are supported on Browser:
```
T:System.Security.Cryptography.RandomNumberGenerator
T:System.Security.Cryptography.IncrementalHash
T:System.Security.Cryptography.SHA1
T:System.Security.Cryptography.SHA256
T:System.Security.Cryptography.SHA384
T:System.Security.Cryptography.SHA512
T:System.Security.Cryptography.SHA1Managed
T:System.Security.Cryptography.SHA256Managed
T:System.Security.Cryptography.SHA384Managed
T:System.Security.Cryptography.SHA512Managed
```
Others throw PNSE and can be marked as unsupported.

Part of https://github.com/dotnet/runtime/issues/41087

### CODE REVIEW

**Comment by vcsjones** [comment]
I think `IncrementalHash.CreateHMAC` could be marked as not supported since Browser doesn't support HMACs. IncrementalHash only works with used with `CreateHash`.

**Comment by MaximLipnin** [comment]
> I think `IncrementalHash.CreateHMAC` could be marked as not supported since Browser doesn't support HMACs. IncrementalHash only works when used with `CreateHash`.

@vcsjones Thanks, addressed

File: src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs [hunk 1]
```diff
@@ -471,7 +495,9 @@ public sealed partial class IncrementalHash : System.IDisposable
         public void AppendData(byte[] data, int offset, int count) { }
         public void AppendData(System.ReadOnlySpan<byte> data) { }
         public static System.Security.Cryptography.IncrementalHash CreateHash(System.Security.Cryptography.HashAlgorithmName hashAlgorithm) { throw null; }
+        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
```
Comment (mdh1418) [inline src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs:498]: Are we not marking this class as unsupported? I have these on the list I'm using

```
M:System.Security.Cryptography.IncrementalHash.GetHashAndReset(System.Span{System.Byte}),System.Security.Cryptography,IncrementalHash,GetHashAndReset(Span<Byte>),0
"M:System.Security.Cryptography.IncrementalHash.CreateHMAC(System.Security.Cryptography.HashAlgorithmName,System.ReadOnlySpan{System.Byte})",System.Security.Cryptography,IncrementalHash,"CreateHMAC(HashAlgorithmName, ReadOnlySpan<Byte>)",0
M:System.Security.Cryptography.IncrementalHash.GetCurrentHash(System.Span{System.Byte}),System.Security.Cryptography,IncrementalHash,GetCurrentHash(Span<Byte>),0
M:System.Security.Cryptography.IncrementalHash.AppendData(System.ReadOnlySpan{System.Byte}),System.Security.Cryptography,IncrementalHash,AppendData(ReadOnlySpan<Byte>),0
M:System.Security.Cryptography.IncrementalHash.get_AlgorithmName,System.Security.Cryptography,IncrementalHash,get_AlgorithmName(),0
M:System.Security.Cryptography.IncrementalHash.GetCurrentHash,System.Security.Cryptography,IncrementalHash,GetCurrentHash(),0
M:System.Security.Cryptography.IncrementalHash.AppendData(System.Byte[]),System.Security.Cryptography,IncrementalHash,AppendData(Byte[]),0
"M:System.Security.Cryptography.IncrementalHash.CreateHMAC(System.Security.Cryptography.HashAlgorithmName,System.Byte[])",System.Security.Cryptography,IncrementalHash,"CreateHMAC(HashAlgorithmName, Byte[])",0
M:System.Security.Cryptography.IncrementalHash.get_HashLengthInBytes,System.Security.Cryptography,IncrementalHash,get_HashLengthInBytes(),0
"M:System.Security.Cryptography.IncrementalHash.TryGetCurrentHash(System.Span{System.Byte},System.Int32@)",System.Security.Cryptography,IncrementalHash,"TryGetCurrentHash(Span<Byte>, Int32)",0
"M:System.Security.Cryptography.IncrementalHash.TryGetHashAndReset(System.Span{System.Byte},System.Int32@)",System.Security.Cryptography,IncrementalHash,"TryGetHashAndReset(Span<Byte>, Int32)",0
"M:System.Security.Cryptography.IncrementalHash.AppendData(System.Byte[],System.Int32,System.Int32)",System.Security.Cryptography,IncrementalHash,"AppendData(Byte[], Int32, Int32)",0
M:System.Security.Cryptography.IncrementalHash.GetHashAndReset,System.Security.Cryptography,IncrementalHash,GetHashAndReset(),0
M:System.Security.Cryptography.IncrementalHash.CreateHash(System.Security.Cryptography.HashAlgorithmName),System.Security.Cryptography,IncrementalHash,CreateHash(HashAlgorithmName),0
```

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACSHA256.cs [hunk 2]
```diff
@@ -4,16 +4,16 @@
 using System;
 using System.Diagnostics;
 using System.Security.Cryptography;
-
 using Internal.Cryptography;
+using System.Runtime.Versioning;
```
Comment (mdh1418) [inline src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACSHA256.cs:8]: Nit: Can we keep System* grouped together?

File: src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs (hunk 1 above)
Comment (MaximLipnin) [reply src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs:498]: Recently @akoeplinger has re-enabled `IncrementalHash` class for Browser in https://github.com/dotnet/runtime/pull/40486

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACSHA256.cs (hunk 2 above)
Comment (MaximLipnin) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACSHA256.cs:7]: Done

**Review by mdh1418 — APPROVED** [review]

**Comment by vcsjones** [comment]
> .. which are supported on Browser:
> T:System.Security.Cryptography.MD5

My understanding is that MD5 support was removed and won't be supported in Browser. https://github.com/dotnet/runtime/pull/40486#issuecomment-685941880.

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AsymmetricKeyExchangeFormatter.cs [hunk 3]
```diff
@@ -1,8 +1,10 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
+using System.Runtime.Versioning;
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AsymmetricKeyExchangeFormatter.cs:3]: Usually there's a blank line between the license header and the first using-import statement.  If it's easy to put them in, please do.

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Internal.Cryptography;
 using System.Buffers;
 using System.Diagnostics;
 using System.Formats.Asn1;
 using System.IO;
+using System.Runtime.Versioning;
 using System.Security.Cryptography.Asn1;
-using Internal.Cryptography;
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs:9]: I believe the using import statements are generally alphabetical-with-System first.  Granted, there are a fair number of exceptions already present in the PR,.. so again it's an "if it's easy to do it alphabetical-but-System-first, please do."

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACSHA1.cs [hunk 5]
```diff
@@ -3,14 +3,15 @@
...
+using System.Runtime.Versioning;
 
 namespace System.Security.Cryptography
 {
     //
     // If you change anything in this class, you must make the same change in the other HMAC* classes. This is a pain but given that the
     // preexisting contract from the .NET Framework locks all of these into deriving directly from HMAC, it can't be helped.
     //
-
+    [UnsupportedOSPlatform("browser")]
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACSHA1.cs:14]: Please restore the blank lines between the meta-comment and the code for all the HMAC classes.

**Comment by MaximLipnin** [comment]
> My understanding is that MD5 support was removed and won't be supported in Browser. [#40486 (comment)](https://github.com/dotnet/runtime/pull/40486#issuecomment-685941880).

Seems that I was confused by the browser api exclusion list containing MD5 type and including it to the build for browser target.
Perhaps some additional clean-up might be done here.

**Comment by MaximLipnin** [comment]
@bartonjs I'll fix those style issues

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AsymmetricKeyExchangeFormatter.cs (hunk 3 above)
Comment (MaximLipnin) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AsymmetricKeyExchangeFormatter.cs:4]: Done

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs (hunk 4 above)
Comment (MaximLipnin) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs:9]: Done

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACSHA1.cs (hunk 5 above)
Comment (MaximLipnin) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACSHA1.cs:15]: Done

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RC2.cs [hunk 6]
```diff
@@ -3,10 +3,12 @@
 
 using Internal.Cryptography;
```
Comment (kjpou1) [inline src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RC2.cs:4]: very very much a nit:  This does not seem to follow all the previous style changes that were made.

Really think these style changes should have been done in a separate PR.

**Review by bartonjs — APPROVED** [review]

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RC2.cs (hunk 6 above)
Comment (MaximLipnin) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RC2.cs]: Done

**Review by kjpou1 — APPROVED** [review]
